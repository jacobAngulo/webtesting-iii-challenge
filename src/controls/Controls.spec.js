// Test away!
import React from "react";
import { render, fireEvent, cleanup } from "react-testing-library";
import renderer from "react-test-renderer";
import "react-testing-library/cleanup-after-each";
import Controls from "./Controls";

describe("<Controls />", () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<Controls />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("is mocking me", () => {
    const mock = jest.fn();

    const result = mock();

    expect(result).toBeUndefined();
    expect(mock).toHaveBeenCalledTimes(1);
  });
});
