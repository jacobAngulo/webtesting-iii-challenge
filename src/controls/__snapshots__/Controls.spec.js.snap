// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="lock-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="close-btn"
  >
    Close Gate
  </button>
</div>
`;
